src/components/web3/api.ts

```diff
@@ -221,6 +241,46 @@ export const web3Prove = async (
   return result;
 };
 
+export const web3SolProve = async (
+  web3Address: string
+): Promise<Web3Authentication> => {
+  if (!web3Address) {
+    throw new Error("not logged into Web3");
+  }
+
+  const nonce = await getNonce();
+  console.log("!!! nonce", nonce);
+  const message: EIP4361Message = {
+    domain: window.location.host,
+    address: web3Address,
+    statement:
+      "Please sign this message so Brave Talk knows that you own this address",
+    uri: window.location.toString(),
+    version: "1",
+    chainId: 1,
+    // HT:https://stackoverflow.com/questions/40031688/javascript-arraybuffer-to-hex/40031979
+    // btoa has some characters not allowed by the EIP-4361 ABNF
+    nonce: nonce,
+    issuedAt: new Date().toISOString(),
```

Comment:
Are we relying upon this time within the backend at all? If so this is potentially modifiable by the client so we'll want to check that verification logic.

TODO for @kdenhartog or @bcaller to look into the impact of malicious times

Reply:
Looks like this one is an issue: https://github.com/brave-intl/subscriptions/pull/367#discussion_r1238032632